### PR TITLE
Simplify null division.

### LINKED ISSUE #3624: The expression `col(a) / null` is not optimized.
**Describe the bug**

**To Reproduce**
On datafusion-cli:
```sql
❯ create table t as select 3 as a;
0 rows in set. Query took 0.015 seconds.
❯ select a / null from t;
+------------+
| t.a / NULL |
+------------+
|            |
+------------+
1 row in set. Query took 0.008 seconds.
❯ explain select a / null from t;
+---------------+-------------------------------------------------+
| plan_type     | plan                                            |
+---------------+-------------------------------------------------+
| logical_plan  | Projection: #t.a / Int64(NULL) AS t.a / NULL    |
|               |   TableScan: t projection=[a]                   |
| physical_plan | ProjectionExec: expr=[a@0 / NULL as t.a / NULL] |
|               |   MemoryExec: partitions=1, partition_sizes=[1] |
|               |                                                 |
+---------------+-------------------------------------------------+
2 rows in set. Query took 0.005 seconds.
```

**Expected behavior**

The expression `col(a) / null` should be optimized to `null`

**Additional context**

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>

# Which issue does this PR close?

Closes #3624.

 # Rationale for this change

# What changes are included in this PR?

Simplify `A / null` and `null / A` to null.
Add tests.

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/optimizer/src/simplify_expressions.rs [hunk 1]
```diff
@@ -753,8 +753,14 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
...
                 op: Divide,
+                right: _,
+            } if is_null(&left) => *left,
+            // null / A --> null
+            BinaryExpr {
+                left: _,
+                op: Divide,
                 right,
-            } if left == right && is_null(&left) => *left,
+            } if is_null(&right) => *right,
```
Comment (alamb) [inline datafusion/optimizer/src/simplify_expressions.rs:763]: Isn't this checking for `A / null` (not `null / A` as the comments suggest)?
 (as in I think the comments are backwards)

**Review by alamb — APPROVED** [review]
Thanks @HaoYang670  -- I think the comments may be mixed up but otherwise 👍

File: datafusion/optimizer/src/simplify_expressions.rs (hunk 1 above)
Comment (HaoYang670) [reply datafusion/optimizer/src/simplify_expressions.rs:763]: Thank you for catching this, I will update.

Comment (HaoYang670) [reply datafusion/optimizer/src/simplify_expressions.rs:763]: Done.

**Pull request merged**